src/main/java/mate/academy/bookstore/serivce/impl/BookServiceImpl.java

```diff
@@ -37,4 +37,19 @@ public BookDto findById(Long id) {
                 .orElseThrow(
                         () -> new EntityNotFoundException("Can't find book by id: " + id));
     }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto requestDto) {
+        if (!bookRepository.existsById(id)) {
+            throw new EntityNotFoundException("Can't update book id: " + id);
+        }
+        Book book = bookMapper.toModel(requestDto);
+        book.setId(id);
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
```

Comment:
we should also check that `id` for existence here, but be careful to not duplicate the code

Reply:
and you did exactly what I warned you about :)
this `if (notExists) {throw ex}` should be a private method for use in update & delete methods. 
Remember, in most cases when you just copy-paste anything in your code - it is a bad practice and you are likely missing an opportunity to extract something to a method and reuse a logic